okhttp-tests/src/test/java/com/squareup/okhttp/WebPlatformUrlTest.java

```diff
@@ -59,46 +59,37 @@
       "Parsing: <http://f:999999/c> against <http://example.org/foo/bar>",
       "Parsing: <#β> against <http://example.org/foo/bar>",
       "Parsing: <http://www.google.com/foo?bar=baz# »> against <about:blank>",
-      "Parsing: <http://[www.google.com]/> against <about:blank>",
```

Comment:
This failing test was one of the motivating cases to do IPv6 in OkHttp itself. When we have a hostname in `[square braces]` we need to confirm that it's an IPv6 address otherwise that's invalid. How do we check if it's an IPv6 address? Either we naively check for at least one `:`, or we fully parse it.